spec/unit/http/session_spec.rb

```diff
@@ -83,4 +83,58 @@ def check_connection?(session, service, ssl_context: nil)
       }.to raise_error(ArgumentError, "Unknown service westbound")
     end
   end
+
+  context 'when resolving using multiple resolvers' do
+    let(:session) { client.create_session }
+
+    it "prefers SRV records" do
+      Puppet[:use_srv_records] = true
+      Puppet[:srv_domain] = "srv.example.com"
+
+      expect(session.resolvers.map(&:class)).to eq([Puppet::HTTP::Resolver::SRV, Puppet::HTTP::Resolver::Settings])
+      expect(session.resolvers[0]).to receive(:resolve).and_return(nil).ordered
+      expect(session.resolvers[1]).to receive(:resolve).and_call_original.ordered
```

Comment:
Since SRV is `resolver[0]`, we want to `.and_call_original` on it, and drop the expectation on the Settings `resolver[1]`